### PR TITLE
Streamline chunk writing

### PR DESCRIPTION
**Pull request by beorn7**
This helps to avoid allocations in the same way we were already doing
it during reading.

@fabxc 

The "fuzz" benchmark doesn't show any significant differences, but in heavy write scenarios (see our MySQL Prometheus), this might actually matter a bit. It's cleaner in any case (consistency between read and write path).

### CODE REVIEW

File: storage/local/persistence.go [hunk 1]
```diff
@@ -67,7 +67,8 @@ const (
 	chunkHeaderFirstTimeOffset = 1
 	chunkHeaderLastTimeOffset  = 9
 	chunkLenWithHeader         = chunkLen + chunkHeaderLen
-	chunkMaxBatchSize          = 64 // How many chunks to load at most in one batch.
+	chunkMaxBatchSize          = 62 // Max no. of chunks to load or write in one batch.
+	// Note that 62 is the largest number of chunks that fit into 64kiB on disk.
```
Comment (juliusv) [inline storage/local/persistence.go]: Because of the external chunk header? Maybe add that to the comment?

**Comment by fabxc** [comment]
:+1:

File: storage/local/persistence.go (hunk 1 above)
Comment (beorn7) [inline storage/local/persistence.go]: Sure.

**Comment by juliusv** [comment]
:+1: otherwise

**Pull request merged**